Partner relations Port to v8

bwrsandman opened the pull request.

hbrunn committed:
[ADD] partner_relations

hbrunn committed:
[FIX] flake8, relative import, unused context initialization

hbrunn committed:
[IMP] context check, make type_id browsable

hbrunn committed:
[RFR] make collecting relation ids simple to override

hbrunn committed:
[IMP] allow to write certain fields on relation view

hbrunn committed:
[RFR] provide a modular way to inject fields into
res_partner_relation_all

hbrunn committed:
[FIX] pep8

Commit:
[FIX] #688, partner_relations: creating equal but non-overlapping relations was not allowed

Conflicts:
	partner_relations/model/res_partner_relation_all.py

NL66278 committed:
[ENH] Module partner_relations:
    - Edit relations in pop-up form instead of through inline tree.

hbrunn committed:
[FIX] coding style issues

hbrunn committed:
[ADD] some demo data

hbrunn committed:
[IMP] put one attribute xml elements on one line

hbrunn committed:
[ADD] support more search operands when search for relations

hbrunn committed:
[ADD] update pot file

hbrunn committed:
[FIX] allow to change partner_id_display

hbrunn committed:
[FIX] logic error

hbrunn committed:
[FIX] search in res.partner.relation.type to support translated searches

hbrunn committed:
[FIX] search both sides of the relation isolated to avoid false
positives

hbrunn committed:
[IMP] support != operator

hbrunn committed:
[FIX] actually do something useful with !=

Commit:
Fix added pep8 conventions

Put boolean operators at end of lines instead of beggining of newlines
Put lambda definition in fields.function def
Remove callit lambda, use function directly

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

bwrsandman committed:
Remove explicit dependencies on web

bwrsandman committed:
Fix v8 api calls

bwrsandman committed:
Replace Table with smart button

Commit:
Fix Read Write, Create and Unlink for realtion_all

Commit:
Hide inactive relations

Commit:
Convert constraints to v8

Commit:
Convert models to v8 api

Commit:
Fix bug with partner_relation any_partner_id field

Commit:
Fix filters in smart button view of relations

foutoucour commented on partner_relations/__openerp__.py:
```diff
@@ -0,0 +1,95 @@
+    "author": "Therp BV,Odoo Community Association (OCA)",
+    "complexity": "normal",
+    "description": """
+Introduction
```
please user OCA template for descriptions:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

foutoucour commented on partner_relations/__openerp__.py:
```diff
@@ -0,0 +1,95 @@
+date.""",
+    "category": "Customer Relationship Management",
+    "depends": [
+        'base',
```
Base is not necessary in v8.
All modules depend from base now.

foutoucour commented on partner_relations/__openerp__.py:
```diff
@@ -0,0 +1,95 @@
+    "depends": [
+        'base',
+    ],
+    "demo": [
```
demo section does not exist anymore in v8. It should be in data section

foutoucour commented on partner_relations/model/res_partner.py:
```diff
@@ -0,0 +1,316 @@
+                                 context=None):
+        '''return the partners' relations as tuple
+        (id, left_partner_id, right_partner_id)'''
+        cr.execute(
```
Could you comment why you use a query instead of ORM here.
I am sure there is a good reason ;)

bwrsandman replied:
I am sure, but that is not my code. Maybe @hbrunn can clarify that point.

hbrunn replied:
there was a time when it was relevant if we found the relation as a left partner or a right partner. Meanwhile, the magic for this is done in `res.partner.relation`, so probably this can be done in the ORM by now. There might be weirdnesses casued by https://github.com/odoo/odoo/pull/5807 though, as both `left_partner_id` and `right_partner_id` are auto_joined.

bwrsandman commented:
There might be a lot of dead code, I need that reviewed.

Also, @alexis-via please let me know what can be done to make this more compatible with the other partner_relation

RoelAdriaans commented on partner_relations/model/res_partner_relation_all.py:
```diff
@@ -0,0 +1,199 @@
+                this.type_selection_id.name_get()[0][1],
+                this.other_partner_id.name,
+            )
+            for this in self.browse(cr, uid, ids, context=context)}
```
Old style browse is used here, gives an error because cr,uid,ids are unknown.

Commit:
Move description into OCA description template

Commit:
Remove redundant base as dependency

Commit:
Remove old style browse

Commit:
Fix domains in smart button

Show selections when partner type is undefined
Apply domain when selecting a type_selection

hbrunn commented on partner_relations/__openerp__.py:
```diff
@@ -0,0 +1,51 @@
+    "js": [
+    ],
+    "css": [
+    ],
```
those two can go away

hbrunn commented on partner_relations/model/res_partner_relation.py:
```diff
@@ -0,0 +1,366 @@
+        store=True,
+    )
+
+    any_partner_id = fields.Many2many(
```
From my understanding, this field is used to allow to search for a partner and see all of her relations. Does this actually work? It's not stored and I don't see a search function

hbrunn commented:
Thanks! :+1:, we'll add tests and some rewrites of legacy code later, I'd say

bwrsandman commented:
@hbrunn I factored out the `id * 10` padding to a variable which can be changed to 2 eventually

I however, noticed that the constraints on left and right partners don't exist in tree view, only in the smart button view and this will have to be done.

Commit:
[IMP] Spanish translation

antespi commented:
@bwrsandman i've PR https://github.com/savoirfairelinux/partner-contact/pull/1 to your PR the Spanish translation, please merge it, in order to append to this PR
:+1:

Commit:
[FIX] Update translation .POT and spanish translation

Commit:
Add french translation

antespi commented on partner_relations/__openerp__.py:
```diff
@@ -0,0 +1,51 @@
+    "author": "Therp BV,Odoo Community Association (OCA)",
+    "complexity": "normal",
+    "category": "Customer Relationship Management",
+    "depends": [
```
Does not depend on base addon?

antespi replied:
Does not depend on base addon?

foutoucour replied:
Base dependency is not required anymore in v8. It is implicitly added actually.

antespi commented on partner_relations/model/res_partner.py:
```diff
@@ -0,0 +1,316 @@
+
+                for relation_type_selection_id in relation_type_selection_ids:
+                    type_id, is_inverse = (
+                        relation_type_selection_id.get_type_from_selection_id()
```
AttributeError: 'int' object has no attribute 'get_type_from_selection_id'
relation_type_selection_ids is a list of integers (old API search method returns a list of integers), so you need to use browse, isn't it?

antespi commented on partner_relations/view/res_partner_relation.xml:
```diff
@@ -0,0 +1,97 @@
+                <tree
+                    string="Partner Relations"
+                    colors="gray:date_end and date_end &lt; current_date or not active;blue:date_start &gt; current_date"
+                    editable="bottom"
```
IMHO, this should be `editable="top"`, because if there are lots of relations, when user clicks to create a new one, apparently nothing happends, user must go to the end of the screen to see an editable line.

antespi commented on partner_relations/view/res_partner_relation_all.xml:
```diff
@@ -0,0 +1,92 @@
+                <tree
+                    string="Partner Relations"
+                    colors="gray:(date_end and date_end &lt; current_date) or not active;blue:date_start &gt; current_date"
+                    editable="bottom"
```
IMHO, this should be `editable="top"`, because if there are lots of relations, when user clicks to create a new one, apparently nothing happends, user must go to the end of the screen to see an editable line.

Commit:
Editable top

Commit:
Use browse in old api functions

rafaelbn commented:
Hi @hbrunn , this module was merged. Could you remove label "needs review" even though it's true, it has a bug.

rafaelbn commented:
https://github.com/OCA/partner-contact/pull/170#issuecomment-141930996
